api/pkg/apis/v1alpha1/providers/target/k8s/k8s.go

```diff
@@ -413,8 +494,23 @@ func (i *K8sTargetProvider) removeDeployment(ctx context.Context, namespace stri
 		}
 	}
 
+	waitErr := i.pollCheck(ctx, true, func() error {
+		_, err := i.Client.AppsV1().Deployments(namespace).Get(ctx, name, metav1.GetOptions{})
+		return err
+	})
+
+	if waitErr != nil {
+		if waitErr == context.DeadlineExceeded {
+			log.ErrorfCtx(ctx, "P (K8s Target Provider): Timeout waiting for deployment %s in namespace %s to be deleted.", name, namespace)
```

Comment:
You still have some logs using "(K8s Target Provider)", please replace them as "(K8s Target)".